include/maliput/common/maliput_throw.h

```diff
@@ -103,75 +103,93 @@ void Throw(const char* condition, const char* func, const char* file, int line)
 }  // namespace common
 }  // namespace maliput
 
-/// Evaluates @p condition and iff the value is false will throw an exception
-/// with a message showing at least the condition text, function name, file,
-/// and line.
-#define MALIPUT_THROW_UNLESS(condition)                                                                               \
+// Helper macros to choose the correct implementation
+#define GET_4TH_ARG(_1, _2, _3, NAME, ...) NAME
+#define GET_3RD_ARG(_1, _2, NAME, ...) NAME
+#define MALIPUT_VALIDATE_MACRO_CHOOSER(...) GET_4TH_ARG(__VA_ARGS__, MALIPUT_VALIDATE_3, MALIPUT_VALIDATE_2)
+#define MALIPUT_THROW_MESSAGE_MACRO_CHOOSER(...) \
+  GET_3RD_ARG(__VA_ARGS__, MALIPUT_THROW_MESSAGE_3, MALIPUT_THROW_MESSAGE_2)
+#define MALIPUT_THROW_UNLESS_MACRO_CHOOSER(...) GET_3RD_ARG(__VA_ARGS__, MALIPUT_THROW_UNLESS_3, MALIPUT_THROW_UNLESS_2)
+
+/// @def MALIPUT_THROW_UNLESS
+/// Evaluates @p condition and iff the value is false will throw a
+/// maliput::common::assertion_error exception with a message showing at least
+/// the condition text, function name, file, and line.
+#define MALIPUT_THROW_UNLESS_2(condition)                                                                             \
```

Comment:
nit: Can you move all helper methods to an internal namespace in this very same file? Given that they are not expected to be used by the user directly.

Reply:
Sure, well spot.